nix/shell.sh

```diff
@@ -41,27 +41,35 @@ function configure {
           -DBUILD_SHARED_LIBS=ON \
           $S2N_CMAKE_OPTIONS \
           -DCMAKE_BUILD_TYPE=RelWithDebInfo
-}
+)}
 
-function build {
+function build {(set -e
     banner "Running Build"
     javac tests/integrationv2/bin/SSLSocketClient.java
     cmake --build ./build -j $(nproc)
     # Build s2n from HEAD
     if [[ -z "${S2N_KTLS_TESTING_EXPECTED}" ]]; then
         $SRC_ROOT/codebuild/bin/install_s2n_head.sh $(mktemp -d)
     fi
-}
+)}
 
-function unit {
+function unit {(set -e
     if [[ -z "$1" ]]; then
-        (cd $SRC_ROOT/build; ctest -L unit -j $(nproc) --verbose)
+        cmake --build build -j $(nproc)
+        ctest --test-dir build -L unit -j $(nproc)
     else
-        (cd $SRC_ROOT/build; ctest -L unit -R $1 -j $(nproc) --verbose)
```

Comment:
Are we removing `--verbose` to simplify the test output? Will we lose some important debugging information?

Reply:
Oh nice catch, I forgot you need "--verbose" for actual useful test output -_- I was just thinking it would get rid of some verbose logging related to ctest. I put it back.